types/ArweaveTransactionID.ts

```diff
@@ -0,0 +1,44 @@
+import { Equatable } from './equatable';
+
+const txIdRegex = /^(\w|-){43}$/;
+export class ArweaveTransactionID implements Equatable<ArweaveTransactionID> {
+  constructor(private readonly transactionId: string) {
+    if (!transactionId.match(txIdRegex)) {
+      throw new Error(
+        'Transaction ID should be a 43-character, alphanumeric string potentially including "-" and "_" characters.',
+      );
+    }
+  }
+
+  [Symbol.toPrimitive](hint?: string): string {
+    if (hint === 'number') {
+      throw new Error('Transaction IDs cannot be interpreted as a number!');
+    }
+
+    return this.toString();
+  }
```

Comment:
i don't believe we'll ever run into this issue since we don't try and case transaction ids to numbers. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toPrimitive)

the error would show if you did something like
````
const id = new ArweaveTransactionID('gh673M0Koh941OIITVXl9hKabRaYWABQUedZxW-swIA')
const idAsNumber = +id; // ERROR WOULD THROW ON THIS CASTING
```

Reply:
actually, i think i have run into this issue, but i cant remember when. Transaction ids that started with a number threw a type error. Believe it was something to do with the gql query and the user wallet.